### ISSUE TITLE
Could not compile due to errors

### ISSUE THREAD

**Issue by ozanmuyes** [cargo-lambda/cargo-lambda#661]
On my MacBook I have Rust installed already

```bash
$ rustup check
stable-x86_64-apple-darwin - Up to date : 1.78.0 (9b00956e5 2024-04-29)
nightly-x86_64-apple-darwin - Update available : 1.80.0-nightly (032af18af 2024-06-02) -> 1.80.0-nightly (7c52d
2db6 2024-06-03)
rustup - Up to date : 1.27.1
```

```bash
$ rustup toolchain list
stable-x86_64-apple-darwin (default)
nightly-x86_64-apple-darwin
```

and did installed `cargo-lambda` via brew;

```bash
$ brew info cargo-lambda
==> cargo-lambda/cargo-lambda/cargo-lambda: stable 1.2.2.x86_64, HEAD
Cargo Lambda is a Cargo subcommand to help you work with AWS Lambda.
https://www.cargo-lambda.info
Installed
/usr/local/Cellar/cargo-lambda/1.2.2.x86_64 (4 files, 28.3MB) *
  Built from source on 2024-06-04 at 17:17:28
From: https://github.com/cargo-lambda/homebrew-cargo-lambda/blob/HEAD/cargo-lambda.rb
License: MIT
==> Dependencies
Required: zig ✔
==> Options
--HEAD
        Install HEAD version
```

I then followed the instructions at [Getting Started page](https://www.cargo-lambda.info/guide/getting-started.html#step-2-create-a-new-project) and ran `cargo lambda new aws_lambda-rust-contact_form --http`. The project was scaffolded.

But when I run `cargo lambda watch` I get the following output;

```bash
$ cargo lambda watch
 INFO starting lambda function function="_" manifest="Cargo.toml" cmd=Exec { prog: "cargo", args: ["run", "--ma
nifest-path", "Cargo.toml", "--color", "auto"] }
   Compiling pin-project-lite v0.2.14
   Compiling itoa v1.0.11
   Compiling futures-core v0.3.30
   Compiling once_cell v1.19.0
   Compiling fnv v1.0.7
   Compiling futures-sink v0.3.30
   Compiling memchr v2.7.2
   Compiling futures-io v0.3.30
   Compiling futures-task v0.3.30
   Compiling pin-utils v0.1.0
   Compiling ryu v1.0.18
   Compiling proc-macro2 v1.0.85
   Compiling libc v0.2.155
   Compiling log v0.4.21
   Compiling slab v0.4.9
   Compiling regex-syntax v0.6.29
   Compiling cfg-if v1.0.0
   Compiling try-lock v0.2.5
   Compiling percent-encoding v2.3.1
   Compiling futures-channel v0.3.30
error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /Users/ozan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.85/src/lib.rs:90:59
   |
90 | #![cfg_attr(any(proc_macro_span, super_unstable), feature(proc_macro_span))]
   |                                                           ^^^^^^^^^^^^^^^

   Compiling regex-syntax v0.8.3
   Compiling want v0.3.1
   Compiling httparse v1.8.0
   Compiling tower-service v0.3.2
   Compiling tinyvec_macros v0.1.1
   Compiling tracing-core v0.1.32
   Compiling form_urlencoded v1.2.1
   Compiling lazy_static v1.4.0
   Compiling tower-layer v0.3.2
   Compiling smallvec v1.13.2
   Compiling sharded-slab v0.1.7
   Compiling tinyvec v1.6.0
   Compiling thread_local v1.1.8
   Compiling base64 v0.22.1
error: expected one of `:`, `;`, `=`, or `where`, found `<`
  --> /Users/ozan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sharded-slab-0.1.7/src/tid.rs:32:1
   |
32 | / lazy_static! {
33 | |     static ref REGISTRY: Registry = Registry {
34 | |         next: AtomicUsize::new(0),
35 | |         free: Mutex::new(VecDeque::new()),
36 | |     };
37 | | }
   | | ^
   | | |
   | | expected one of `:`, `;`, `=`, or `where`
   | |_while parsing this item list starting here
   |   the item list ends here
   |
   = note: this error originates in the macro `__lazy_static_internal` which comes from the expansion of the ma

   Compiling unicode-bidi v0.3.15
   Compiling encoding_rs v0.8.34
error[E0046]: not all trait items implemented, missing: `Target`, `deref`
  --> /Users/ozan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sharded-slab-0.1.7/src/tid.rs:32:1
   |
32 | / lazy_static! {
33 | |     static ref REGISTRY: Registry = Registry {
34 | |         next: AtomicUsize::new(0),
35 | |         free: Mutex::new(VecDeque::new()),
36 | |     };
37 | | }
   | |_^ missing `Target`, `deref` in implementation
   |
   = note: this error originates in the macro `__lazy_static_internal` which comes from the expansion of the ma
   = help: implement the missing item: `type Target = /* Type */;`
   = help: implement the missing item: `fn deref(&self) -> &<Self as Deref>::Target { todo!() }`

   Compiling base64 v0.21.7
For more information about this error, try `rustc --explain E0554`.
error: could not compile `proc-macro2` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error[E0609]: no field `free` on type `REGISTRY`
   --> /Users/ozan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sharded-slab-0.1.7/src/tid.rs:150:14
    |
150 |             .free
    |              ^^^^ unknown field
    |
    = note: available field is: `__private_field`

error[E0609]: no field `next` on type `REGISTRY`
   --> /Users/ozan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sharded-slab-0.1.7/src/tid.rs:161:35
    |
161 |                 let id = REGISTRY.next.fetch_add(1, Ordering::AcqRel);
    |                                   ^^^^ unknown field
    |
    = note: available field is: `__private_field`

error[E0609]: no field `free` on type `REGISTRY`
   --> /Users/ozan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sharded-slab-0.1.7/src/tid.rs:191:42
    |
191 |             let mut free_list = REGISTRY.free.lock().unwrap_or_else(PoisonError::into_inner);
    |                                          ^^^^ unknown field
    |
    = note: available field is: `__private_field`

Some errors have detailed explanations: E0046, E0609.
For more information about an error, try `rustc --explain E0046`.
error: could not compile `sharded-slab` (lib) due to 5 previous errors
ERROR command exited code=101

^C INFO terminating lambda scheduler
 INFO terminating lambda function function="_"
```

In a nutshell I can't build and run `cargo-lambda` scaffolded project.

Edit: Here's the `Cargo.toml`;

```toml
[package]
name = "aws_lambda-rust-contact_form"
version = "0.1.0"
edition = "2021"

# Starting in Rust 1.62 you can use `cargo add` to add dependencies
# to your project.
#
# If you're using an older Rust version,
# download cargo-edit(https://github.com/killercup/cargo-edit#installation)
# to install the `add` subcommand.
#
# Running `cargo add DEPENDENCY_NAME` will
# add the latest version of a dependency to the list,
# and it will keep the alphabetic ordering for you.

[dependencies]
lambda_http = "0.11.1"

tokio = { version = "1", features = ["macros"] }

```

**Comment by ozanmuyes** [comment]
I think the issue stems from ["sharded-slab"](https://github.com/hawkw/sharded-slab/issues/92).

**Comment by ozanmuyes** [comment]
The issue was due to **corrupted cargo registry**. Deleting `$HOME/.cargo/registry` and re-building the app did the trick. Sorry for the possible inconvenience.
